analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/SymbolicValueAssertions.cs

```diff
@@ -0,0 +1,125 @@
+{
+    public static class SymbolicValueExtensions
+    {
+        public static SymbolicValueAssertions Should(this SymbolicValue instance) =>
+            new(instance);
+    }
+
+    public class SymbolicValueAssertions : ReferenceTypeAssertions<SymbolicValue, SymbolicValueAssertions>
+    {
+        protected override string Identifier => "symbolicValue";
+
+        public SymbolicValueAssertions(SymbolicValue subject) : base(subject) { }
+
+        public AndWhichConstraint<SymbolicValueAssertions, SymbolicConstraint> HaveOnlyConstraint(SymbolicConstraint expected, string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(Subject != null)
+                .FailWith("The {context:symbolicValue} is null and can not have constraint {0}.", expected)
+                .Then
+                .Given(() => Subject.AllConstraints.ToList())
+                .ForCondition(actual => actual.Count > 0)
+                .FailWith("Expected {context:symbolicValue} to have constraint {0}{reason}, but SymbolicValue has no constraints.", expected)
+                .Then
+                .ForCondition(actual => actual.Count == 1)
+                .FailWith("Expected {context:symbolicValue} to have only constraint {0}{reason}, but SymbolicValue has {1} constraints.", _ => expected, x => x.OrderBy(x => x.ToString()))
+                .Then
+                .ForCondition(actual => actual[0] == expected)
+                .FailWith("Expected {context:symbolicValue} to have constraint {0}{reason}, but SymbolicValue has {1} constraint.",
+                    _ => expected,
+                    actual => actual[0]);
+            var matched = Subject.AllConstraints.First(x => x == expected);
+            return new(this, matched);
+        }
+
+        public AndConstraint<SymbolicValueAssertions> HaveNoConstraints(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(Subject == null || !Subject.AllConstraints.Any())
+                .FailWith("Expected {context:symbolicValue} to have no constraints{reason}, but {0} constraints were found.", Subject?.AllConstraints.OrderBy(x => x.ToString()));
+            return new(this);
+        }
+
+        public AndConstraint<SymbolicValueAssertions> HaveOnlyConstraints(params SymbolicConstraint[] expected)
+            => HaveOnlyConstraints(expected, string.Empty);
+
+        public AndConstraint<SymbolicValueAssertions> HaveOnlyConstraints(IEnumerable<SymbolicConstraint> expected, string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(expected != null && expected.Any())
+                .FailWith("Expected constraints is empty. Use HaveNoConstraints() instead.")
```

Comment:
This could backfire in parameterized tests when you have a mixture of cases with some constraints and others without. We could add an overload with a `bool allowEmpty` option or remove this condition.

Reply:
Let's wait and see if we'll need it. We can always improve later.